### PR TITLE
Stories/ethnicity race update logic

### PR DESCRIPTION
**Pull request by eanders**
This completes https://www.pivotaltracker.com/story/show/169846478

### CODE REVIEW

File: spec/models/grda_warehouse/tasks/client_cleanup_spec.rb [hunk 1]
```diff
@@ -338,24 +266,106 @@
...
+          destination_client.update(col => 0)
+          source_1.update(col => 1, DateUpdated: 1.day.ago)
+          source_2.update(col => 0, DateUpdated: 2.days.ago)
+          @cleanup.update_client_demographics_based_on_sources
+          destination_client.reload
+          expect(destination_client[col]).to eq(1)
+        end
+
+        it 'uses newest known race value' do
+          destination_client.update(Gender: 1)
```
Comment (awisema) [inline spec/models/grda_warehouse/tasks/client_cleanup_spec.rb]: `Gender` should be `col`?

File: spec/models/grda_warehouse/tasks/client_cleanup_spec.rb [hunk 2]
```diff
@@ -338,24 +266,106 @@
...
+          destination_client.update(col => 0)
+          source_1.update(col => 1, DateUpdated: 1.day.ago)
+          source_2.update(col => 0, DateUpdated: 2.days.ago)
+          @cleanup.update_client_demographics_based_on_sources
+          destination_client.reload
+          expect(destination_client[col]).to eq(1)
+        end
+
+        it "uses newest known #{col} value" do
+          destination_client.update(Gender: 1)
```
Comment (awisema) [inline spec/models/grda_warehouse/tasks/client_cleanup_spec.rb]: Same

File: app/models/grda_warehouse/tasks/client_cleanup.rb [hunk 3]
```diff
@@ -219,13 +221,77 @@ def choose_best_gender dest_attr, source_clients
...
+    private def race_columns
+      @race_columns ||= GrdaWarehouse::Hud::Client.race_fields.map(&:to_sym) - [:RaceNone]
+    end
+
+    def choose_best_ethnicity dest_attr, source_clients
+      # Most recent 0 or 1 if no 0 or 1 use the most recent value
+      known_values = [0, 1]
+      # Sort in reverse chronological order (newest first)
+      sorted_source_clients = source_clients.sort_by.sort{|a, b| b[:DateUpdated] <=> a[:DateUpdated]}
+      col = :Ethnicity
```
Comment (awisema) [inline app/models/grda_warehouse/tasks/client_cleanup.rb]: If the column is set as a variable anyway, can we DRY anything else up by extracting out `choose_best_column_value` or such like?

Comment (eanders) [reply app/models/grda_warehouse/tasks/client_cleanup.rb]: Yes, we could, though the logic is only the same for a few of them, and historically people have wanted different logic for each type.  I'm inclined to leave it for now, but this is good to note for later.

File: spec/models/grda_warehouse/tasks/client_cleanup_spec.rb (hunk 1 above)
Comment (eanders) [reply spec/models/grda_warehouse/tasks/client_cleanup_spec.rb]: Thanks for catching these.

**Pull request merged**